### PR TITLE
Constrained umac grown function

### PR DESCRIPTION
**Pull request by esclapez**
Add a `create_constrained_umac_grown` function allowing to fill umac ghost faces with a divergence-free staggered velocity field. The old `create_umac_grown` is kept for retro-compatibility at this point.
Note that `create_constrained_umac_grown` uses `InterpFromCoarseLevel` with a `face_divfree_interp` interpolater and requires a `GpuBndryFuncFab` for mixed index type.

### CODE REVIEW

File: Utils/hydro_utils.H [hunk 1]
```diff
@@ -153,6 +153,34 @@ create_umac_grown (int lev, int nGrow, amrex::BoxArray& fine_grids,
...
+      amrex::Array4<amrex::Real> const& /*dummy*/,
+      const int /*dcomp*/,
+      const int /*numcomp*/,
+      amrex::GeometryData const& /*geom*/,
+      const amrex::Real /*time*/,
+      const amrex::BCRec* /*bcr*/,
+      const int /*bcomp*/,
+      const int /*orig_comp*/) const
+   { } 
+};
```
Comment (cgilet) [inline Utils/hydro_utils.H:184]: @esclapez If we're not supposed to ever get here, maybe an abort should be here, jic?

**Comment by esclapez** [comment]
@cgilet Mixed GpuBndryFuncFab are now in AMReX development [AMReX-Codes/amrex#2238](https://github.com/AMReX-Codes/amrex/pull/2238) so this is probably ready to merge if you want to move forward on resolving Godunov/EBGodunov discrepancies.

**Comment by cgilet** [comment]
@esclapez IAMR has several tests that die with this error:
amrex::Abort::0::FCFillExtDirDummy is a dummy placeholder and should not be used in combination with BCType::ext_dir !!!

I see it for cases where the fine grid touches an inflow boundary, slip wall, and no slip and outflow (not sure if it's just one of no slip wall or outflow causing the crash or both since the fine grid touches both and I haven't tracked that down yet). 

For cases where the fine grid touches an inflow boundary, since Dirichlet BCs are considered to lie exactly on the boundary, shouldn't the Fill functions in IAMR be getting called here? 

For slip walls/no slip walls/outflow, does PeleLM work with fine grid touching the boundary?

**Comment by esclapez** [comment]
It shouldn't matter what the actual BC is since the function set internally all the umac BC type to foextrap: 

https://github.com/AMReX-Codes/AMReX-Hydro/pull/17/commits/906393906b14a7f6c2d1da5be19e62da7642cb47#diff-8fb4441d7667a5793e88db3f940e5a050a3887659d81aba8cecd187832eb7331R204-R208

I'll try to reproduce the error to see what's the problem.

**Pull request merged**